packages/esm-patient-list-app/src/api/queries.ts

```diff
@@ -113,3 +117,40 @@ export function useRemovePatientsFromOfflinePatientListMutation() {
     }
   });
 }
+
+export function usePatientListDetails(patientListUuid: string): [OpenmrsCohort, KeyedMutator<{ data: OpenmrsCohort }>] {
+  if (patientListUuid) {
+    const {
+      data: cohortDetails,
+      error,
+      mutate,
+    } = useSWR<{ data: OpenmrsCohort }, Error>(`${cohortUrl}/cohort/${patientListUuid}`, openmrsFetch);
```

Comment:
Careful here! You are calling a hook (`useSWR`) inside an `if`. See [this](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level) for details.

Can the `patientListUuid` even be falsy? If not, you can just remove the `if` entirely.

I'd also avoid throwing inside hooks (there are exceptions, of course, but for data fetching hooks based on SWR, the pattern is that any error is returned, not thrown).

All in all, you can probably condense the entire hook to the following: 

```ts
export function usePatientListDetails(patientListUuid: string) {
  return useSWR<{ data: OpenmrsCohort }, Error>(`${cohortUrl}/cohort/${patientListUuid}`, openmrsFetch);
}
```